### ISSUE TITLE
Renewable items

### ISSUE THREAD

**Issue by TheDeviantCrafter** [Glitchfiend/BiomesOPlenty#2159]
### Overview

I went through all the non-renewable items, and want to suggest adding recipes. I tried to keep things easy to implement.

-----------------

**Crafting:**

- 2 rotten flesh + 2 netherrack in a 2x2 = 4 flesh blocks
- Sand + iron nugget = orange sand
- Sand + bone meal = white sand
- Sand + flint = black sand
- 2 blaze powder + 2 netherrack in a 2x2 = 4 brimstone
- 8 brimstone around a fire charge = brimstone fumarole
- 8 rotten flesh around a bucket = blood bucket

**Stonecutter recipes:**

- Flesh block in a stonecutter = porous flesh, flesh tendons, flesh tendons strand, eyebulb, hair, and pus bubble
- Cobweb in a stonecutter = webbing or hanging cobweb
- Brimstone in stonecutter = brimstone bud, brimstone cluster
- Blackstone in stonecutter = blackstone spines

**Bone meal:**
- Applying bone meal to a torchflower, blackstone bulb, cattail, watergrass or reed drops another one as an item like a rose bush
- If a bramble block is not connected to any other bramble, it can grow like a sapling and instantly become one of the fully grown bramble plants that generate during world gen (which means you can use the same worldgen object rather than writing growth code like a chorus plant)
- Applying bone meal to grass in the appropriate biome has a chance to grow shrubs, dead grass, desert grass, dune grass, tundra shrub, clover, sprout, barley and sea oats
(For consistency's sake, if there is a grass-like block, using bone meal on grass should create it. Even shrubs in nether overgrowth biomes - players can bring their own grass.)
- Applying bone meal to glowworm silk causes it to grow down by 1 block

**Other:**
- Rose quartz blocks grow a small rose quartz bud when random ticked. Buds grow 1 size larger when random ticked.
- Killing a spider with bane of arthopods has a chance to drop a spider egg (0.5% per enchantment level)
- Smelting a `<tag:items:forge:storage_blocks/salt>` gives dried salt. This would require another mod like Mekanism to add a salt block, but I can't think of a better mechanic. Some way to get salt from drying the ocean seems way out of scope for BOP.
- Breaking a large clover without shears drops a small clover (applying bone meal to a small clover makes a 2x2 huge clover, so this would make both renewable)

### Why would this feature be useful?

Biomes O Plenty has some nice building blocks, but most of them aren't renewable which makes them difficult to work with at scale. I don't want to dismantle an entire biome to build a black sandstone house.

**Comment by TheDeviantCrafter** [comment]
Side note, here are a few renewables that are already in game:
- All flowers are renewable
- Applying bone meal to a toadstool or glowshroom makes a giant mushroom
- Glowing moss follows vanilla moss rules
- Redwood, fir and umbran saplings have 2x2 variants if you want to repopulate those biomes with big trees
- Hanging cobwebs will turn into hanging cobweb strands if you place another under it, so there's no reason to give strands a recipe. Same with glowworm silk.

**Comment by Forstride** [comment]
I'm open to the idea of some recipes that make sense like Rotten Flesh -> Flesh blocks, but ones like crafting for sands won't happen.  Those are already renewable through Wandering Villager trades, which is the only way sand is renewable in vanilla.

A lot of these just feel kinda forced for the sake of renewability, and I'm already not a huge fan of the idea that everything should be renewable through farming, so...I'd only want to add it in cases where it makes sense and feels appropriate in-world.

**Comment by BlueTangsRock** [comment]
I agree that the resources in Biomes O' Plenty definitely should have better methods of renewability. Especially the plants; I've been using bonemeal across lots of the modded biomes and only a small fraction of the foliage appears to be renewable this way.

Edit: I've been playing on the 1.18.2 version. It seems like the more recent versions address these issues, so that explains why I've been experiencing these types of issues, lol. I'm happy that these issues had been worked on and fixed in the newer versions of the mod. :)

**Comment by Forstride** [comment]
I'm not sure we can really do anything about regrowing normal foliage on grass since that's hardcoded into the grass block itself and isn't extendable without overriding the grass bonemeal feature (Although that does mean you could make a datapack for it if you want to).  I did make it so Mossy Black Sand grows Sprouts when bonemealed though, but for Bushes and small Clovers, you'll have to rely on the Wandering Trader for that (Although maybe I'll make it so Huge Clover Petals drop small Clovers when broken without shears).

As for other stuff, I added quite a few recipes/interactions for renewability in 1.20.4.  For example, Flesh blocks adjacent to Blood now slowly grow fleshy decorations on them (Pus Bubbles, Hair, Eyebulbs, Flesh Tendons underneath).  I think pretty much everything is renewable in some form between in-world interactions and the Wandering Trader, except for Blood, since we can't do the Dripstone thing that Water/Lava has.  I do still want to add some better method for Brimstone renewal though, and eventually make Bramble bonemealable as well.

**Comment by BlueTangsRock** [comment]
Perhaps for 1-block tall foliage (flowers, grasses, bushes, clovers, and sprouts), using bonemeal on a specific plant could grow more of the same plant around itself in a small radius, regardless of biome? A similar mechanic works for flowers on Bedrock Edition, so it might be possible to implement a similar mechanic here.

**Comment by Forstride** [comment]
That's not needed for small flowers since using bonemeal on grass blocks already regrows them (It was bugged in some older versions but it's been fixed for quite some time now).  Could maybe work for Bushes, but I'd rather try to find another solution that's more in line with other foliage blocks first.

**Comment by TheDeviantCrafter** [comment]
> That's not needed for small flowers since using bonemeal on grass blocks already regrows them 

This works for some plants but it's fairly inconsistent. All of the plants I mentioned above were not renewable in 1.20.1 at least based on my testing.

**Comment by Lucky-56** [comment]
I just wanna throw dead saplings in here, similar to how alex's caves handles it with branches of the tree combined with some fertilizer
![thornwood](https://github.com/Glitchfiend/BiomesOPlenty/assets/55949451/06e79781-dfd9-45c8-a0a5-fbf51d5f0ff5)
just mentioning this because I only found a singular tree that dropped me 6 saplings which eventually died down to none but was able to give me ~4 stacks of logs
